cmd/cloudFoundryDeploy.go

```diff
@@ -0,0 +1,855 @@
+}
+
+func deployCfNative(deployConfig deployConfig, config *cloudFoundryDeployOptions, additionalEnvironment []string, cmd command.ExecRunner) error {
+
+	deployStatement := []string{
+		deployConfig.DeployCommand,
+	}
+
+	if len(deployConfig.AppName) > 0 {
+		deployStatement = append(deployStatement, deployConfig.AppName)
+	}
+
+	if len(deployConfig.DeployOptions) > 0 {
+		deployStatement = append(deployStatement, deployConfig.DeployOptions...)
+	}
+
+	if len(deployConfig.ManifestFile) > 0 {
+		deployStatement = append(deployStatement, "-f")
+		deployStatement = append(deployStatement, deployConfig.ManifestFile)
+	}
+
+	if len(config.DeployDockerImage) > 0 && config.DeployType != "blue-green" {
+		deployStatement = append(deployStatement, "--docker-image", config.DeployDockerImage)
+	}
+
+	if len(config.DockerUsername) > 0 && config.DeployType != "blue-green" {
+		deployStatement = append(deployStatement, "--docker-username", config.DockerUsername)
+	}
+
+	if len(deployConfig.SmokeTestScript) > 0 {
+		deployStatement = append(deployStatement, deployConfig.SmokeTestScript...)
+	}
+
+	if len(config.CfNativeDeployParameters) > 0 {
+		deployStatement = append(deployStatement, strings.FieldsFunc(config.CfNativeDeployParameters, func(c rune) bool {
+			return c == ' '
+		})...)
+	}
+
+	stopOldAppIfRunning := func(_cmd command.ExecRunner) error {
+
+		if config.KeepOldInstance && config.DeployType == "blue-green" {
+			oldAppName := deployConfig.AppName + "-old"
+			_ = oldAppName
+
+			var buff bytes.Buffer
+
+			oldOut := _cmd.Stdout
+			_ = oldOut
+			_cmd.Stdout(&buff)
+
+			defer func() {
+				_cmd.Stdout(log.Writer())
+			}()
+
+			err := _cmd.RunExecutable("cf", "stop", oldAppName)
+
+			if err != nil {
+
+				cfStopLog := buff.String()
+
+				if !strings.Contains(cfStopLog, oldAppName+" not found") {
+					return fmt.Errorf("Could not stop application %s. Error: %s", oldAppName, cfStopLog)
+				}
+				log.Entry().Infof("Cannot stop application '%s': %s", oldAppName, cfStopLog)
+
+			} else {
+				log.Entry().Infof("Old application '%s' has been stopped.", oldAppName)
+			}
+		}
+
+		return nil
+	}
+
+	return cfDeploy(config, deployStatement, additionalEnvironment, stopOldAppIfRunning, cmd)
+}
+
+func getManifest(name string) (cloudfoundry.Manifest, error) {
+	return cloudfoundry.ReadManifest(name)
+}
+
+func getAppNameOrFail(config *cloudFoundryDeployOptions) (string, error) {
+
+	if len(config.AppName) > 0 {
+		return config.AppName, nil
+	}
+	if config.DeployType == "blue-green" {
+		return "", fmt.Errorf("Blue-green plugin requires app name to be passed (see https://github.com/bluemixgaragelondon/cf-blue-green-deploy/issues/27)")
+	}
+	if len(config.Manifest) == 0 {
+		return "", fmt.Errorf("Manifest file not provided in configuration. Cannot retrieve app name")
+	}
+	fileExists, err := fileUtils.FileExists(config.Manifest)
+	if err != nil {
+		return "", err
+	}
```

Comment:
I prefer not to swallow errors. Hence we should not _ here IMO. It is unlikely that we get an error here, but experience teaches us that we have to deal also with unlikely cases. And in case such an unlikely case happens the troubleshooter does not have the details.

Reply:
I am fine with that. But I do suggest to wrap the error then to add the context where it happened.

Reply:
Oh, I see I already said that within the suggestion.

Reply:
This is the same BTW for all other cases of `FileExists()` where I commented on. I.e. I am fine with handling the error, but then it needs to be wrapped or I am convinced that you would not achieve the goal of debugging a failure.